concourse/client/model.py

```diff
@@ -76,11 +79,39 @@ def __init__(self, raw:dict, concourse_api, name:str):
             warning('Pipeline did not contain resource definitions: {p}'.format(p=name))
             raise ValueError()
         self.resources = map(lambda r: Resource(r, self), resources)
+        jobs = self.raw.get('jobs', None)
+        if not jobs:
+            warning(f'Pipeline {name} did not contain job definitions')
+            raise ValueError()
+        self.jobs = (Job(job, self) for job in jobs)
```

Comment:
this is a generator comprehension. which means that the result may only be iterated over _once_. I do not think this is a good idea (hence my suggestion to use list comprehension earler..)